TheaterSchedule.BLL/Services/ImageService.cs

```diff
@@ -1,16 +1,188 @@
 ﻿using System;
+using System.Linq;
 using System.Collections.Generic;
 using System.Text;
+using System.IO;
+using System.Threading.Tasks;
+using System.Reflection;
+using System.Drawing;
+using System.Drawing.Imaging;
+using Microsoft.Extensions.Caching.Memory;
+using Entities.Models;
 using TheaterSchedule.BLL.Interfaces;
+using TheaterSchedule.DAL.Interfaces;
+using TheaterSchedule.BLL.DTO;
+
 namespace TheaterSchedule.BLL.Services
 {
-   public class ImageService: IImageService
+    public class ImageService : IImageService
     {
+        private IImageRepository imageRepository;
+        private IMemoryCache cache;
+
+        private int cacheExpirationMinutes = 20;
+
+        public ImageService(IImageRepository imgRepository, IMemoryCache memoryCache)
+        {
+            imageRepository = imgRepository;
+            cache = memoryCache;
+        }
+
         public string ImageToBase64(byte[] image)
         {
             string base64Image = Convert.ToBase64String(image);
             return base64Image;
         }
-    }
 
+        public string GeMimeTypeFromImageByteArray(byte[] byteArray)
+        {
+            try
+            {
+                using (MemoryStream stream = new MemoryStream(byteArray))
+                {
+                    using (Image image = Image.FromStream(stream))
+                    {
+                        return ImageCodecInfo.GetImageEncoders().First(codec => codec.FormatID == image.RawFormat.Guid).MimeType;
+                    }
+                }
+            }
+            catch (ArgumentException)
+            {
+                return "text/plain";
+            }
+        }
+
+        public async Task<ImageBase64DTO> LoadPerformanceMainImageBase64Async(int id)
+        {
+            ImageBase64DTO imageDTO = null;

```

Comment:
don't duplicate logic of loading bytes from repository
call method LoadPerformanceMainImageBytesAsync and convert bytes to base64 string